### ISSUE TITLE
runs and installs 10.6.0, fails when running sudo node-install -v 8 in bash

### ISSUE THREAD

**Issue by JadedBlueEyes** [audstanley/NodeJs-Raspberry-Pi#17]
```
Traceback (most recent call last):
  File "/bin/node-install", line 241, in <module>
    installDependencies()
  File "/bin/node-install", line 68, in installDependencies
    o, e = Popen(['pip', 'install', 'requests'], stdout=PIPE, stderr=PIPE).communicate()
  File "/usr/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1024, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

```

~~it seems it's failing to create/access a directory it needs.~~
edit: it's a error with pip. as @Kryten0807 said:

> The problem is that the "Installing Python Pip..." step failed silently. The node-install script attempts to install python2-pip, but in Raspbian Stretch lite that package is called python-pip.

the full code is:

```

pi@node-pi-001:~ $ sudo bash
root@node-pi-001:/home/pi# sudo wget -O - https://raw.githubusercontent.com/audstanley/NodeJs-Raspberry-Pi/master/Install-Node.sh | bash
--2018-07-13 15:54:28--  https://raw.githubusercontent.com/audstanley/NodeJs-Raspberry-Pi/master/Install-Node.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.16.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.16.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1689 (1.6K) [text/plain]
Saving to: ‘STDOUT’

-                                          100%[========================================================================================>]   1.65K  --.-KB/s    in 0.003s

2018-07-13 15:54:30 (481 KB/s) - written to stdout [1689/1689]

--2018-07-13 15:54:32--  https://nodejs.org/dist/latest-v10.x/node-v10.6.0-linux-armv6l.tar.gz
Resolving nodejs.org (nodejs.org)... 104.20.22.46, 104.20.23.46, 2400:cb00:2048:1::6814:172e, ...
Connecting to nodejs.org (nodejs.org)|104.20.22.46|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 17284155 (16M) [application/gzip]
Saving to: ‘node-v10.6.0-linux-armv6l.tar.gz’

node-v10.6.0-linux-armv6l.tar.gz           100%[========================================================================================>]  16.48M  3.45MB/s    in 5.6s

2018-07-13 15:54:38 (2.94 MB/s) - ‘node-v10.6.0-linux-armv6l.tar.gz’ saved [17284155/17284155]

update-alternatives: using /opt/nodejs/bin/node to provide /usr/bin/node (node) in auto mode
update-alternatives: using /opt/nodejs/bin/npm to provide /usr/bin/npm (npm) in auto mode
rm: cannot remove 'node-install': No such file or directory
--2018-07-13 15:55:10--  https://raw.githubusercontent.com/audstanley/NodeJs-Raspberry-Pi/master/node-install
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.16.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.16.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10826 (11K) [text/plain]
Saving to: ‘node-install’

node-install                               100%[========================================================================================>]  10.57K  --.-KB/s    in 0.004s

2018-07-13 15:55:10 (2.44 MB/s) - ‘node-install’ saved [10826/10826]

root@node-pi-001:/home/pi# sudo node-install -v 8
                                  ;
                                  +++
                                  +++
                                  +++
   ''++''       :;;',        ,+++;+++    ''++''
:++++++++++: ;;;;;;;''''  '++++'+++++ :++++++++++:
+++.    .++' ';;;;;;;;''  +++     +++ '++. '' .+++
+++.    .++' '';;;;;;;;'  +++     +++ '++. '' ''
+++.    .++' ''';;;;;;;;  '++++'+++++ :+++++,
:          ,   '';;;,        ,+++;       ''++'.
                   #+''#';+''+'
                   ';;;+#'+;;'
                    .###''@#@
                    '@++@@'+#@
                  :'@'''@'''''+
                   ##@@'''#@+#@
                   .'''#''@''#
                     '#'''#'


Updating Repositories...
Need to install some dependencies, this may take a few minutes...
Installing Python Pip...
Installing Python Requests Library...
Traceback (most recent call last):
  File "/bin/node-install", line 241, in <module>
    installDependencies()
  File "/bin/node-install", line 68, in installDependencies
    o, e = Popen(['pip', 'install', 'requests'], stdout=PIPE, stderr=PIPE).communicate()
  File "/usr/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1024, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
root@node-pi-001:/home/pi# exit
```

**Comment by audstanley** [comment]
Thank you so much for the report, I'm literally testing right now. :-)
 Fresh install of stretch, and I had no errors, though I ran the update, so I'm going to reinstall stretch, and NOT run the desktop prompted upgrade to see if i can replicate.  I noticed also that Popen(['pip', 'install', 'requests'], i forgot to add a '-y' flag, I fixed that and it looks like stretch comes with the requests library (after I ran the upgrade of course).  Time to reflash and try again. I'm on a Pi2 so this is a long process.

much appreciated.  This helps out a lot.
